benchmarks/src/main/kotlin/ChannelProducerConsumerBenchmarkWorker.kt

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2016-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.selects.select
+import java.util.concurrent.ForkJoinPool
+import java.util.concurrent.Phaser
+
+abstract class ChannelProducerConsumerBenchmarkWorker(private val withSelect: Boolean,
```

Comment:
Let's rename it `ChannelProducerConsumerBenchmarkIteration` and pass `producers` and `consumers` as parameters so that it has a simple `run()` function.